packages/jbrowse-cli/src/commands/add-connection.ts

```diff
@@ -0,0 +1,317 @@
+          { exit: 40 },
+        )
+    } else configContents.connections.push(connectionConfig)
+
+    this.debug(`Writing configuration to file ${configPath}`)
+    await fsPromises.writeFile(
+      configPath,
+      JSON.stringify(configContents, undefined, 2),
+    )
+
+    this.log(
+      `${idx !== -1 ? 'Overwrote' : 'Added'} connection "${name}" ${
+        idx !== -1 ? 'in' : 'to'
+      } ${configPath}`,
+    )
+  }
+
+  async checkLocation(location: string) {
+    let manifestJson: string
+    try {
+      manifestJson = await fsPromises.readFile(
+        path.join(location, 'manifest.json'),
+        {
+          encoding: 'utf8',
+        },
+      )
+    } catch (error) {
+      this.error(
+        'Could not find the file "manifest.json". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 50 },
+      )
+    }
+    let manifest: { name?: string } = {}
+    try {
+      manifest = JSON.parse(manifestJson)
+    } catch (error) {
+      this.error(
+        'Could not parse the file "manifest.json". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 60 },
+      )
+    }
+    if (manifest.name !== 'JBrowse') {
+      this.error(
+        '"name" in file "manifest.json" is not "JBrowse". Please make sure you are in the top level of a JBrowse 2 installation.',
+        { exit: 70 },
+      )
+    }
+  }
+
+  async resolveURL(location: string, check = true) {
+    let locationUrl: URL | undefined
+    try {
+      locationUrl = new URL(location)
+    } catch (error) {
+      this.error('The location provided is not a valid URL', { exit: 80 })
+    }
+    if (locationUrl) {
+      let response
+      try {
+        if (check) {
+          response = await fetch(locationUrl, { method: 'HEAD' })
+        }
+        if (!response || response.ok) return locationUrl.href
+      } catch (error) {
+        // ignore
+        this.error('Unable to fetch from URL', { exit: 90 })
+      }
+    }
+    return this.error(`Could not resolve to a URL: "${location}"`, {
+      exit: 100,
+    })
+  }
+
+  async readJsonConfig(location: string) {
+    let locationUrl: URL | undefined
+    try {
+      locationUrl = new URL(location)
+    } catch (error) {
+      // ignore
+    }
+    if (locationUrl) {
+      const response = await fetch(locationUrl)
+      return response.json()
+    }
+    return fsPromises.readFile(location, { encoding: 'utf8' })
+  }
+
+  determineConnectionType(url: string, config: string | undefined) {
+    if (path.basename(url) === 'hub.txt') return 'UCSCTrackHubConnection'
+    if (url.includes('jbrowse/data')) return 'JBrowse1Connection'
+    if (config && this.isValidJSON(config)) return 'custom'
+    return this.error(
+      `Unable to determine a specific connection from URL given.\nPlease specify a type with --type.\nIf you want a custom type, please provide the config object with --config`,
```

Comment:
Probably should use `os.EOL` instead of `\n` for Windows support.

Reply:
Does a console log really have to worry about os.EOL?